src/lightning/pytorch/utilities/compile.py

```diff
@@ -122,6 +123,7 @@ def _maybe_unwrap_optimized(model: object) -> "pl.LightningModule":
         return from_compiled(model)
     if isinstance(model, pl.LightningModule):
         return model
+    _check_mixed_imports(model)
```

Comment:
why not include this check outside _maybe_unwrap_optimized()?  Seems unrelated to unwrapping.

Reply:
Just for the convenince of adding it to a single place. Currently, users who mix things up get the error just below